api/v1beta1/disruption_schedule_types.go

```diff
@@ -0,0 +1,87 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023 Datadog, Inc.
+
+package v1beta1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+func init() {
+	SchemeBuilder.Register(&DisruptionSchedule{}, &DisruptionScheduleList{})
+}
+
+//+kubebuilder:object:root=true
+
+// DisruptionSchedule is the Schema for the disruptions API
+// +kubebuilder:resource:shortName=disch
+// +kubebuilder:subresource:status
+type DisruptionSchedule struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              DisruptionScheduleSpec   `json:"spec,omitempty"`
+	Status            DisruptionScheduleStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// DisruptionScheduleList contains a list of DisruptionSchedule
+type DisruptionScheduleList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []DisruptionSchedule `json:"items"`
+}
+
+// DisruptionScheduleSpec defines the desired state of DisruptionSchedule
+type DisruptionScheduleSpec struct {
+	// +kubebuilder:validation:MinLength=0
+	// +kubebuilder:validation:Required
+	// +ddmark:validation:Required=true
+	// The schedule in Cron format, see https://en.wikipedia.org/wiki/Cron.
+	Schedule string `json:"schedule"`
+
+	//+kubebuilder:validation:Minimum=0
+	// Optional deadline in seconds for starting the disruption if it misses scheduled
+	// time for any reason.  Missed disruption executions will be counted as failed ones.
+	// +nullable
+	StartingDeadlineSeconds *int64 `json:"startingDeadlineSeconds,omitempty"`
```

Comment:
We can allow for a literal duration, using the DisruptionDuration type, so users can specify in a more human readable string, such as "1m", "300s", "30m", rather than having to convert to seconds. We also should come up with a non-zero default, and thus don't need this to be a pointer